src/command.rs

```diff
@@ -12,14 +12,14 @@ pub fn find_selected_version(command: &str) -> Option<String> {
         .map(|version| version.to_owned())
 }
 
-pub fn find_system_bin(command: &str) -> Option<PathBuf> {
-    let shim_dir = config::shim_dir();
-    let path = env::var("PATH").expect("env var PATH is not defined");
-    let paths: Vec<_> = env::split_paths(&path)
-        .filter(|p| p != &shim_dir)
-        .collect();
+fn realpath_equal(a: &Path, b: &Path) -> bool {
+    fs::canonicalize(&a).unwrap() == fs::canonicalize(&b).unwrap()
+}
 
-    paths.iter()
-        .find(|p| p.join(command).exists())
+pub fn find_system_bin(command: &str, system_path: &str, current_exe: &Path) -> Option<PathBuf> {
+    env::split_paths(&system_path)
         .map(|p| p.join(command))
+        .filter(|p| p.exists())
+        .filter(|p| !realpath_equal(&p, &current_exe))
```

Comment:
In this case, we're going to calling `fs::canonicalize(...)` on `current_exe` a bunch of times. We don't really need this.